.github/workflows/build.yml

```diff
@@ -71,6 +77,9 @@ jobs:
           channelId: live
 
   stage:
+    permissions:
+      checks: write
```

Comment:
How were this permissions determined?

Reply:
For `pull-requests: write`, the Firebase action adds comments to PRs as seen by the above comment (this knowledge base also included it https://github.com/step-security/secure-workflows/blob/main/knowledge-base/firebaseextended/action-hosting-deploy/action-security.yml).

As for `checks: write`, the build failed without it, I'm assuming since it adjusts the results of the `build/deploy` job.

Reply:
Thanks for the details!